pkg/controller/deactivation/deactivation_controller_test.go

```diff
@@ -0,0 +1,305 @@
+			mur := murtest.NewMasterUserRecord(t, username, murtest.Account("cluster1", *basicTier), murtest.ProvisionedMur(murProvisionedTime), murtest.UserIDFromUserSignup(userSignupFoobar))
+			r, req, cl := prepareReconcile(t, mur.Name, basicTier, mur, userSignupFoobar)
+			// when
+			res, err := r.Reconcile(req)
+			// then
+			require.NoError(t, err)
+			require.False(t, res.Requeue, "requeue should not be set")
+			require.True(t, res.RequeueAfter == 0, "requeueAfter should not be set")
+			assertThatUserSignupDeactivated(t, cl, username, true)
+		})
+
+		// the time since the mur was provisioned exceeds the deactivation timeout period for the 'other' tier
+		t.Run("usersignup should be deactivated - other tier (60 days)", func(t *testing.T) {
+			// given
+			murProvisionedTime := &metav1.Time{Time: time.Now().Add(-time.Duration(expectedDeactivationTimeoutOtherTier*24) * time.Hour)}
+			mur := murtest.NewMasterUserRecord(t, username, murtest.Account("cluster1", *otherTier), murtest.ProvisionedMur(murProvisionedTime), murtest.UserIDFromUserSignup(userSignupFoobar))
+			r, req, cl := prepareReconcile(t, mur.Name, otherTier, mur, userSignupFoobar)
+			// when
+			res, err := r.Reconcile(req)
+			// then
+			require.NoError(t, err)
+			require.False(t, res.Requeue, "requeue should not be set")
+			require.True(t, res.RequeueAfter == 0, "requeue should not be set")
+			assertThatUserSignupDeactivated(t, cl, username, true)
+		})
+
+	})
+
+	t.Run("failures", func(t *testing.T) {
+
+		// cannot find NSTemplateTier
+		t.Run("unable to get NSTemplateTier", func(t *testing.T) {
+			// given
+			murProvisionedTime := &metav1.Time{Time: time.Now()}
+			mur := murtest.NewMasterUserRecord(t, username, murtest.Account("cluster1", *basicTier), murtest.ProvisionedMur(murProvisionedTime), murtest.UserIDFromUserSignup(userSignupFoobar))
+			r, req, cl := prepareReconcile(t, mur.Name, mur, userSignupFoobar)
+			// when
+			_, err := r.Reconcile(req)
+			// then
+			require.Error(t, err)
+			require.Contains(t, err.Error(), `nstemplatetiers.toolchain.dev.openshift.com "basic" not found`)
+			assertThatUserSignupDeactivated(t, cl, username, false)
+		})
+
+		// cannot find NSTemplateTier
+		t.Run("NSTemplateTier without deactivation timeout", func(t *testing.T) {
+			// given
+			murProvisionedTime := &metav1.Time{Time: time.Now()}
+			mur := murtest.NewMasterUserRecord(t, username, murtest.Account("cluster1", *noDeactivationTier), murtest.ProvisionedMur(murProvisionedTime), murtest.UserIDFromUserSignup(userSignupFoobar))
+			r, req, cl := prepareReconcile(t, mur.Name, noDeactivationTier, mur, userSignupFoobar)
+			// when
+			res, err := r.Reconcile(req)
+			// then
+			require.NoError(t, err)
+			require.False(t, res.Requeue, "requeue should not be set")
+			require.True(t, res.RequeueAfter == 0, "requeue should not be set")
+			assertThatUserSignupDeactivated(t, cl, username, false)
+		})
```

Comment:
this test doesn't cover any failure and looks like a duplication of https://github.com/codeready-toolchain/host-operator/pull/281/files#diff-61785d47f3d452a843fb690e18cb6eefR88-R100